bin/validate.js

```diff
@@ -167,11 +203,16 @@ function validateImageExtension(filename) {
 function requireImageDecode(fileName) {
   // Test https://golang.org/pkg/image/png/#Decode and https://golang.org/pkg/image/jpeg/#Decode are possible
   return new Promise((resolve, reject) => {
-    exec(`bin/validate-image ${fileName}`, (err) => {
-      if (err) {
-        reject(err);
+    const childProcess = spawn('bin/validate-image', [fileName]);
+    childProcess.on('error', (error) => {
+      reject(error);
+    });
+    childProcess.on('close', (code) => {
```

Comment:
Are `error` and `close` mutually exclusive? Otherwise the promise gets resolved and/or rejected (twice).

Reply:
Yes, `error` and `close` are mutually exclusive, the code promises only one outcome: if an error comes up, it rejects the promise. On a successful close, it resolves it. If it closes with issues, it's also rejected.